src/main/java/homework/seven/ArrayValueCalculator.java

```diff
@@ -0,0 +1,30 @@
+package homework.seven;
+
+public class ArrayValueCalculator {
+    String[][] array = new String[4][4];
+
+    public int doCalc(String[][] array) throws ArraySizeException, ArrayDataException {
+        int suma = 0;
+        this.array = array;
+        if (this.array.length > 4) {
+            throw new ArraySizeException();
+        } else {
+            for (int i = 0; i < 4; i++) {
+                if (this.array[i].length > 4) {
+                    throw new ArraySizeException("index "+i);
+                } else {
+                    for (int j = 0; j < 4; j++) {
```

Comment:
What happen if we have 5 elements?

Reply:
will show the following error. [Array size is index 2 = 3 should be 4]